### ISSUE TITLE
AudioMixer Groups

### ISSUE THREAD

**Issue by CraigASanford** [resonance-audio/resonance-audio-unity-sdk#19]
Are Unity Mixer Groups supported on the default ResonanceAudioMixer?

I have a few groups setup on it, and sliders to adjust it down, pretty standard issue volume controls for music/sfx/master, etc. The only one that seems to have any effect is the Master group.

I have a fire that is playing resonanceaudiosource with the audiosource component output set to the SFX group. Turn down the SFX group and it does nothing (despite seeing the levels change in the editor). But gets quieter when I turn down the master volume group.

**Comment by anokta** [comment]
Hi,

Due to limitations in Unity's audio framework, the `Resonance Audio Renderer` plugin (in the `ResonanceAudioMixer` asset) currently does not rely on the incoming audio input to produce its output. This means two things:

1. The mixing chain of an `AudioSource` (including the volume controls) will be ignored by the master renderer - except for the post effects that are applied in the component level (e.g., adding an [AudioLowPassFilter](https://docs.unity3d.com/ScriptReference/AudioLowPassFilter.html) component to a source by enabling `Spatialize Post Effects` parameter).
2. Likewise, all the additional effects prior to `Resonance Audio Renderer` will not be applied. However, it is possible to add effects/attenuation to the whole mix _after_  the spatialization, as you have observed with the master volume group.

We've been discussing with Unity to resolve this limitation as soon as possible. In the meantime, for more sophisticated use cases, we recommend the [Resonance Audio SDK for Wwise](https://github.com/resonance-audio/resonance-audio-wwise-sdk), where such grouping/mixing features are fully supported.

Hope this helps!

**Comment by AnnaMy82** [comment]
Hi, has there been an update on when this fix might be available?

**Comment by andybak** [comment]
I'd also like an update - or more info on workarounds for common workflows without bringing in whole new applications like WWise.

**Comment by zacksettel** [comment]
Agreed that using Wwise is not a viable solution.  I too would like this issue to be resolved soon, since it currently is prohibiting the use of Asset bundles for my mobile app.

**Comment by maxwellasmith** [comment]
Currently looking for a solution as well. Has anyone had any luck finding a work around within Unity?

**Comment by AnnaMy82** [comment]
The only hack I've found works is to use Animator tracks on the Timeline
and animate wether a sound source is active and the volume level.
Alternatively add custom code to an Audio track on the Timeline to control
volume.

**Comment by zacksettel** [comment]
The only work-around I know of is to temporarily relocate the ResonanceAudioMixer outside of the Assets directory(and subdirectories) before creating the asset bundle.   Embarrassing.

**Comment by beardordie** [comment]
This is a problem even in the recently-released Unity 3D Game Kit (Lite, at least which is what I'm testing). The game kit's Pause Menu has an audio slider for master (works great) and "sfx" which is actually a slider for the "Spatial" mixer group, but due to the Resonance Audio Renderer effect, the volume slider has no effect on the volume. Bypass that renderer, and the slider works as expected. Or, you can enable "Allow Wet Mixing (causes higher memory usage)" on the Resonance Audio Renderer, and the slider works. I kept the Wet level at the default -46.50 dB. I'm not sure if the higher memory usage referenced is too high to consider using this setting in a project.

**Comment by ssfrr** [comment]
I'm pretty sure this is the same issue - is there any way to have multiple ResonanceAudioRenderer effects on different mixer groups? I have two different groups of spatialized objects and I'd like to cross-fade between the groups, but creating two ResonanceAudioRenderers creates a bunch of clicks. I get that the renderers are not using the regular input pipeline, but it would be great if they still only received input from AudioSources that are assigned to them, and also were able to mix their outputs, so that you could have this kind of arrangement.

**Comment by ericdrobinson** [comment]
Just ran into the exact same scenario posted by @beardordie. It was _super_ confusing to continue to hear audio when the attenuation was set to basically mute the group. At the very least you could add a warning about this issue to the Editor view for the _Resonance Audio Renderer_ in the inspector...

**Comment by pan-athen** [comment]
> We've been discussing with Unity to resolve this limitation as soon as possible. In the meantime, for more sophisticated use cases, we recommend the [Resonance Audio SDK for Wwise](https://github.com/resonance-audio/resonance-audio-wwise-sdk), where such grouping/mixing features are fully supported.
> 
> Hope this helps!

Have in mind that Wwise is a paid solution and it might be above budget for some projects, while Resonance is free.

**Comment by zacksettel** [comment]
Glad to see some action in this thread.  Would  anyone know if there’s any further development planned for Resonance in Unity?

Zack

**Comment by ericdrobinson** [comment]
Not looking good. The current release (v2.0.0 on Package Manager) [has been deprecated](https://docs.unity3d.com/Manual/com.unity.google.resonance.audio.html) and will be removed in Unity 2020.1. This repo in particular hasn't seen any action in ~2 years... Even the [main repo](https://github.com/resonance-audio/resonance-audio) hasn't seen much over that same span of time.

¯\\\_(ツ)\_/¯

**Comment by joshcamas** [comment]
Yep, looks like Resonance is just dead, leaving very little options for unity audio at all. Beautiful.
